rust/agama-dbus-server/src/network/model.rs

```diff
@@ -16,6 +16,8 @@ use thiserror::Error;
 use uuid::Uuid;
 use zbus::zvariant::Value;
 
+use super::dbus::ControllerSettings;
```

Comment:
Yep, it is specific to DBUS but mainly because of the need to pass just one argument (the ports or the options) It probably could be better written and definitely it smells badly.